client/src/entrypoints/draftail.js

```diff
@@ -40,3 +40,33 @@ const plugins = [
 ];
 
 plugins.forEach(draftail.registerPlugin);
+
+class DraftailRichTextArea {
+  constructor(options) {
+      this.options = options;
+  }
+
+  render(container, name, id) {
+    const options = this.options;
+
+    return {
+      getValue() {
+        return input.value;
+      },
+      getState() {
+        return input.value;
+      },
+      setState(state) {
+        // We must initialise it here because Draftail doesn't provide a way to change state after initialisation
```

Comment:
Hmm. As far as I can think of, nothing in StreamField's current or planned feature set requires us to be able to change the state of an already-initialised widget. (For example, copying blocks will only require us to read the state of an existing block and write that to a newly-initialised one.)

However, the getState / setState API is designed on the assumption that this _is_ possible, and if that's not the case then we should probably revise the API (i.e. have the `render` method accept an initial state) rather than have Draftail pretend it supports the API as currently designed (and then have it come back to bite us when we're using that API more intensively)

I have to wonder how much work it would be to update Draftail to accommodate this, but either way that doesn't change the basic facts: if it's non-trivial to get Draftail to co-operate, then chances are there are going to be other Django widgets in the wild that are similarly difficult or impossible to fit into telepath's world view. It's in our interests to keep telepath as permissive as possible when it comes to integrating 'difficult' widgets - we don't want "rewrite your Django widgets in telepath" to be the new "rewrite your Django widgets in React" :-)

With that in mind, I think the way forward is:

* Add an `initialState` argument to the `render` method in telepath's widget API, and update the StreamField infrastructure to use that rather than calling setState on newly-initialised widgets
* Keep setState as an optional method of the API. Widgets can still use it in their own internals (as PageChooser does, for example) and outside code can call it _if_ it knows that it's dealing with a widget type that implements it. Container blocks like StructBlock can still implement setState recursively, but it will only be functional if all descendant blocks also implement it.

It's quite possible that in future, requirements will come up that _do_ require setState to be implemented more generally - for example, I can imagine people wanting a general-purpose "if field X is filled in, then prefill field Y" mechanism as per the title/slug fields. Establishing setState as an optional non-essential method now will make it easier to reason about those interactions and the limitations of them when they do come up (i.e. it won't work for prefilling a rich text field until we go ahead and implement that properly in Draftail).